tests/sensor_definition/test_reaction_force_sensor.py

```diff
@@ -77,6 +77,12 @@ def test_full_boundary_reaction(dim: int, degree: int) -> None:
     assert force_top == pytest.approx(-1 * force_bottom)
     # checking equal forces on sides
     assert force_left == pytest.approx(force_bottom)
+    # checking report metadata
+    # TODO Figure out how to identify which boundary is applied
+    assert fem_problem.sensors.ReactionForceSensor.report_metadata()["surface"] == "boundary"
+    assert fem_problem.sensors.ReactionForceSensor2.report_metadata()["surface"] == "boundary"
+    assert fem_problem.sensors.ReactionForceSensor3.report_metadata()["surface"] == "boundary"
+    assert fem_problem.sensors.ReactionForceSensor4.report_metadata()["surface"] == "boundary"
```

Comment:
I cannot find a way with the current implementation to identify which boundary is considered. An option would be having a dictionary somewhere with the pairs (string, function) to define the boundaries, but I am not sure where does it make the most sense

Reply:
I comment not answering your question, the `add_sensor` sensor functoins has an optional argument `name`, which you can use to give better sensor names, than ReactionForceSensor, ReactionForceSensor2, ReactionForceSensor3, ...

Now to your question. I am not sure what you are trying to do here. As far as I know, the sensor has the definition of the surface implemented, its just not very homan readable, but I also do not see how this can be achieved for more complex surfaces. If I understad correctly you are looking for a description of the applied surface, or would the name of the function, used to apply the surface be sufficient?

Reply:
I am not sure how to solve this issue either. The problem is that if I want to be able to rebuild the sensor from the metadata,  I need some description of the surface. Right now, the sensor only stores a function object that is generated at run-time, always with the name "boundary" and whose behaviour is determined by a call to the boundary conditions builders (for example, `surface=setup.boundary_at_top()`-->`return plane_at(top_surface)` --> `return boundary`). If I could get the name "boundary_at_top", that would be sufficient to rebuild the sensor, but the sensor itself has no memory of calling that function, as it belongs to setup.

One option would be adding some kind of string indicating the surface to consider and associate that to the corresponding call to the setup functions ("top" -->`surface=new_setup.boundary_at_top()`), but this would only work for very simple cases with those functions defined in the experimental setup, not for arbitrary surfaces. Alternatively, I could try to save the surface itself, but then we would have to find a way to map the saved surface to the new mesh where the sensor is rebuilt.

Reply:
even specialized cases, I would assume will be implemented as function in the specific experiment. maybe at some point including arguments, but these could also be included in the meta data right? So to find a way of saving "function name" + possible arguments, sounds doable.

Reply:
Ok, then I will change the input to "surface" to a dictionary with the name and arguments. Something like 
```
surface={
    "function": function_name,
    "args":{}
}
```

Reply:
Solved in commit [e7c91ae](https://github.com/BAMresearch/FenicsXConcrete/pull/108/commits/e7c91aeef78dcc19cc00861625af65cafdedb26f)